staging/src/k8s.io/client-go/tools/cache/fifo.go

```diff
@@ -235,8 +235,8 @@ func (f *FIFO) List() []interface{} {
 func (f *FIFO) ListKeys() []string {
 	f.lock.RLock()
 	defer f.lock.RUnlock()
-	list := make([]string, 0, len(f.items))
-	for key := range f.items {
+	list := make([]string, 0, len(f.queue))
```

Comment:
This isn't correct change - queue contains only "not yet consumed' items.

Reply:
However - this isn't correct change here.

In particular in Delete method - you may return something from items without touching queue:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/cache/fifo.go#L218

So let's revert the change to fifo - and I can then approve the changes to delta_fifo.

Reply:
yes, done. The changes to fifo had rolled back. 

Besides, would it be better to keep the internal keys of `queue` and `items` consistent in fifo?

Reply:
Good question - we would need to examine better the consequences and current usage.

Reply:
@wojtek-t  please review https://github.com/kubernetes/kubernetes/pull/105166 , which is the PR about keeping keys consistent between `queue` and `items` in fifo.